R/check_forecasts.R

```diff
@@ -125,49 +143,86 @@ check_forecasts <- function(data) {
   }
   data[, InternalDuplicateCheck := NULL]
 
-  # check whether there is a model column present. And if not, state what that means
-  if (!("model" %in% colnames(data))) {
-    msg <- c(
-      msg,
-      paste(
-        "There is no column called `model` in the data.",
-        "scoringutils therefore thinks that all forecasts come from the same model" #nolint
-      )
-    )
-    data[, model := "Unspecified model"]
-  }
 
-  # some checks whether there are the same number of quantiles, samples
+  # check whether there are the same number of quantiles, samples --------------
   data[, InternalNumCheck := length(prediction), by = forecast_unit]
   n <- unique(data$InternalNumCheck)
   if (length(n) > 1) {
     warnings <- c(
       warnings,
       paste0(
-        "Some forecasts have different numbers of rows (e.g. quantiles or samples). ", #nolint
+        "Some forecasts have different numbers of rows (e.g. quantiles or samples). ", # nolint
         "scoringutils found: ", paste(n, collapse = ", "),
         ". This is not necessarily a problem, but make sure this is intended."
       )
     )
   }
   data[, InternalNumCheck := NULL]
 
-  # get available unique values per model for the different columns
+
+  # store info so it can be accessed by the user -------------------------------
+  out[["cleaned_data"]] <- data
+
+  # info to be printed
+  out[["info"]] <- c(
+    paste0(
+      "Forecasts are for a `", target_type, "` target ",
+      "using a `", prediction_type, "` prediction format."
+    ),
+    paste0(
+      "The unit of a single forecast is defined by `",
+      paste(forecast_unit, collapse = "`, `"), "`. ",
+      "If this is not as intended, please DELETE UNNECESSARY columns or add new ones." # nolint
+    )
+  )
+
+  # available unique values per model for the different columns
   cols <- forecast_unit[forecast_unit != "model"]
-  check[["unique_values"]] <-
+  out[["unique_values"]] <-
     data[, lapply(.SD, FUN = function(x) length(unique(x))), by = "model"]
 
-  check[["messages"]] <- unlist(msg)
-  check[["warnings"]] <- unlist(warnings)
-  check[["errors"]] <- unlist(errors)
+  # forecast infos
+  out[["forecast_unit"]] <- forecast_unit
+  out[["target_type"]] <- target_type
+  out[["prediction_type"]] <- prediction_type
 
-  class(check) <- c("scoringutils_check", "list")
+  out[["messages"]] <- unlist(messages)
+  out[["warnings"]] <- unlist(warnings)
+  out[["errors"]] <- unlist(errors)
 
-  return(check)
+
+  # generate messages, warnings, errors ----------------------------------------
+  if (length(messages) > 0) {
+    msg <- paste("The following messages were produced when checking inputs:",
+      messages,
+      sep = "\n"
+    )
+    message(msg)
+  }
+  if (length(warnings) > 0) {
+    msg <- paste("The following warnings were produced when checking inputs:",
+      warnings,
+      sep = "\n"
+    )
+    warning(msg)
+  }
+  if (length(errors) > 0) {
+    print("Duplicate forecasts:")
```

Comment:
Add a `find_duplicates()` function and use this to check for duplicates adding a error if found and the advice to call `find_duplicates()` to id them (as discussed)